### PR TITLE
Trustapi support

### PR DESCRIPTION
**Pull request by dpereskokov**
Fixes: #88 #103

### CODE REVIEW

File: packages/rpc/package.json [hunk 1]
```diff
@@ -63,7 +63,9 @@
     "axios": "^0.19.0",
     "bignumber.js": "^9.0.0",
     "class-transformer": "^0.2.3",
-    "reflect-metadata": "^0.1.13"
+    "reflect-metadata": "^0.1.13",
+    "@trustwallet/trust-api": "0.1.0",
```
Comment (vikmeup) [inline packages/rpc/package.json]: `@trustwallet/trust-api` feels redundant, let's try `api` ?

File: packages/rpc/src/trust-api/Query.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export class Query {
+    rpcUrl: string;
+
+    constructor(rpcUrl: string) {
```
Comment (vikmeup) [inline packages/rpc/src/trust-api/Query.ts]: can we do just `url` ? because it's not rpc url

File: packages/rpc/src/trust-api/TrustRPC.spec.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    let rpc: TrustRPC;
+    const coin = CoinType.cosmos;
+
+    beforeAll(() => {
+        require('dotenv').config({ path: __dirname + '/./.env' });
+        rpc = new TrustRPC(getEnv('TRUST_RPC_URL'));
+    });
+
+    it('should get Cosmos price', async () => {
+        const price = await rpc.getPrices(coin, 'USD');
```
Comment (vikmeup) [inline packages/rpc/src/trust-api/TrustRPC.spec.ts]: can add `USD` to `Currency` types?

File: packages/rpc/src/trust-api/TrustRPC.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import 'reflect-metadata';
+import axios from 'axios';
+import { plainToClass } from 'class-transformer';
+import BigNumber from 'bignumber.js';
+import { Query } from './Query';
+import { CoinType } from '@trustwallet/types/lib/CoinType';
+import Utils from '@trustwallet/trust-api';
+import { PriceResponse } from './models/PriceInfo';
+
+export class TrustRPC {
```
Comment (vikmeup) [inline packages/rpc/src/trust-api/TrustRPC.ts]: maybe `API` or `Client` ?

File: packages/rpc/src/trust-api/TrustRPC.spec.ts (hunk 3 above)
Comment (dpereskokov) [reply packages/rpc/src/trust-api/TrustRPC.spec.ts]: This one is just for tests. But yeah, I'l fix it.

File: packages/rpc/src/trust-api/TrustRPC.ts (hunk 4 above)
Comment (dpereskokov) [reply packages/rpc/src/trust-api/TrustRPC.ts]: Client looks better

File: packages/types/src/CoinType.ts [hunk 5]
```diff
@@ -1,3 +1,7 @@
+export enum FiatCoinType {
+    USD = 'USD',
```
Comment (vikmeup) [inline packages/types/src/CoinType.ts]: should we have separate file for it?

File: packages/rpc/src/trust-client/Query.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+export class Query {
+    rpcUrl: string;
+
+    constructor(rpcUrl: string) {
```
Comment (vikmeup) [inline packages/rpc/src/trust-client/Query.ts]: just `url` ?

File: packages/rpc/src/trust-client/TrustClient.test.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        rpc = new TrustClient(getEnv('TRUST_RPC_URL'));
+    });
+
+    beforeEach(function() {
+        spyOn(axios, 'get').and.returnValue({ data: { backend: {} } });
+        spyOn(axios, 'post').and.returnValue({ data: { docs: [], price: '1' } });
+    });
+
+    it('should get Cosmos price', async () => {
+        const addrToBeCalled = `${getEnv('TRUST_RPC_URL')}/prices`;
```
Comment (vikmeup) [inline packages/rpc/src/trust-client/TrustClient.test.ts]: `TRUST_RPC_URL ` => `TRUST_URL` ?

File: packages/rpc/src/trust-client/TrustClient.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+                {
+                    contract: addr,
+                },
+            ],
+        };
+
+        const respRaw = await axios.post(url, body);
+        const resp = plainToClass(PriceResponse, respRaw.data);
+        const coins = resp.docs;
+        const cosmos = coins.find(coin => coin.contract === addr);
```
Comment (vikmeup) [inline packages/rpc/src/trust-client/TrustClient.ts]: `cosmos` rename to `result`?

File: packages/rpc/src/trust-client/models/PriceInfo.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { Transform, Type } from 'class-transformer';
+import BigNumber from 'bignumber.js';
+
+export class Blockchain {
```
Comment (vikmeup) [inline packages/rpc/src/trust-client/models/PriceInfo.ts]: do we need this class `Blockchain` . ?

File: packages/rpc/src/trust-client/TrustClient.ts (hunk 8 above)
Comment (leoneparise) [reply packages/rpc/src/trust-client/TrustClient.ts]: +1

File: packages/rpc/src/trust-client/models/PriceInfo.ts (hunk 9 above)
Comment (dpereskokov) [reply packages/rpc/src/trust-client/models/PriceInfo.ts]: It's just migrated from the platform here. It's not used. Let's remove it

**Pull request merged**